core/ir/opnd_shared.c

```diff
@@ -2407,6 +2408,10 @@ opnd_compute_address(opnd_t opnd, dr_mcontext_t *mc)
 const char *
 get_register_name(reg_id_t reg)
 {
+    dcontext_t *dcontext = get_thread_private_dcontext();
```

Comment:
Is this the right way of getting the global dcontext?
(It behaves as expected in the tests, I'm just not sure I'm taking every corner case into consideration)

Reply:
+ get_t_p_d can return NULL; dr_get_isa_mode handles that; call get_t* inside that call to avoid someone using the dcontext accidentally?  Or add a comment that it can be NULL.

+ Wondering about the overhead of calling get_t* in every little IR function: some of which are inlined as perf matters on all the getter and setter routines.  But maybe only a small % of these routines will need dc?  Is it all the register conversion ones, but none of the instr or opnd getter or setters?  I would add a comment about this to remind us when we expand for #1684.

+ Re: the prior conversation on an explicit parameter to be consistent w/ decode() and other key routines taking in dcontext: goes back to how many we'd have to change.  There is some precedence for IR routines to call get_t_p_d: instr_get_eflags, opnd_create_far_abs_addr, others.  Maybe include a mention of this in the aforementioned comment discussing tradeoffs/options.

+ Any issues w/ the calling thread being different from the target thread/dcontext: maybe.  Xref #6690.

Reply:
Thank you for the thorough explanation!
Added comments.
And put get_thread_private_dcontext() as argument to get_isa_mode().
Done.